specification/resources/resource-manager/Microsoft.Authorization/stable/2023-04-01/policyDefinitionVersions.json

```diff
@@ -0,0 +1,899 @@
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK - Returns an array of policy definition versions.",
+            "schema": {
+              "$ref": "#/definitions/PolicyDefinitionVersionListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/listPolicyDefinitionVersions": {
+      "post": {
+        "tags": [
+          "PolicyDefinitionVersions"
+        ],
+        "operationId": "PolicyDefinitionVersions_ListAll",
+        "summary": "Lists all policy definition versions within a subscription.",
+        "description": "This operation lists all the policy definition versions for all policy definitions within a subscription.",
+        "x-ms-examples": {
+          "List all policy definition versions at subscription": {
+            "$ref": "./examples/listAllPolicyDefinitionVersions.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK - Returns an array of policy definition versions.",
+            "schema": {
+              "$ref": "#/definitions/PolicyDefinitionVersionListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/policyDefinitions/{policyDefinitionName}/versions/{version}": {
+      "put": {
+        "tags": [
+          "PolicyDefinitionVersions"
+        ],
+        "operationId": "PolicyDefinitionVersions_CreateOrUpdate",
+        "summary": "Creates or updates a policy definition in a subscription.",
+        "description": "This operation creates or updates a policy definition in the given subscription with the given name.",
+        "x-ms-examples": {
+          "Create or update a policy definition version": {
+            "$ref": "./examples/createOrUpdatePolicyDefinitionVersion.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "policyDefinitionName",
```

Comment:
>policyDefinitionName [](http://example.com/codeflow?start=21&length=20)

Considering factoring common parameter definitions to the "parameters" section below.